action/action.go

```diff
@@ -24,15 +24,8 @@ var (
 
 // Action is the action can be Executed in protocols. The method is added to avoid mistakenly used empty interface as action.
 type Action interface {
```

Comment:
the reason why `actionPyaload` is internal type is to give the authority to create action payload to this package, so other packages cannot create actionPyaload

Reply:
and the reason `Action` exist, is because i want when other packages take in an action as input, like 
```
func (p *Protocol) handleTransfer(ctx context.Context, act action.Action, sm protocol.StateManager) (*action.Receipt, error)
```
It takes a concrete type as function param instead use `interface{}` to confuse user.  And because `actionPyaload` is internal type because the reason above, i defined `Action` just to avoid using `interface{}` so people can make mistakes.